src/components/edit/editLineForm/MakbilaList.tsx

```diff
@@ -33,77 +35,171 @@ export const MakbilaMenu = (props: Props) => {
   const handleDeleteParallel = async (index: number) => {
-    const updatedParallels = [...parallels];
-    updatedParallels.splice(index, 1);
-    await saveParallelsToDb(updatedParallels, 'המקבילה נמחקה בהצלחה!', 'שגיאה במחיקת המקבילה');
+    const tractate = getTractate(currentMishna);
+    const chapter = getChapter(currentMishna);
+    
+    if (!tractate || !chapter) {
+      showError('לא ניתן למחוק - מידע המשנה לא זמין עדיין');
+      return;
+    }
+
+    const parallelToDelete = parallels[index];
+    if (!parallelToDelete) {
+      showError('מקבילה לא נמצאה');
+      return;
+    }
+
+    try {
+      // Backend now handles both same-mishna and cross-mishna reciprocal deletion
+      await LineService.deleteParallel(tractate, chapter, currentMishna.mishna, currentLineNumber, parallelToDelete);
+      
+      // Refresh the UI by refetching the mishna
+      const updatedMishna = await PageService.getMishna(tractate, chapter, currentMishna.mishna);
+      const currentLine = updatedMishna.lines?.find(line => line.lineNumber === currentLineNumber);
+      // PageService.getMishna() already converts sublinePairs to selectedSublineIndices
+      const updatedParallels = currentLine?.parallels || [];
+      
+      onUpdateInternalSources(updatedParallels);
+      showSuccess('המקבילה נמחקה בהצלחה!');
+    } catch (error) {
+      console.error('Error deleting parallel:', error);
+      showError('שגיאה במחיקת המקבילה');
+    }
+  };
+
+  // Handle editing a parallel (using NEW granular operation)
+  const handleEditParallel = async (editedParallel: iParallelLink) => {
+    const tractate = getTractate(currentMishna);
+    const chapter = getChapter(currentMishna);
+    
+    if (!tractate || !chapter) {
+      showError('לא ניתן לעדכן - מידע המשנה לא זמין עדיין');
+      return;
+    }
+
+    // Find the original parallel being edited
+    const indexToReplace = parallels.findIndex(p => 
+      p.tractate === editingParallel?.tractate &&
+      p.chapter === editingParallel?.chapter &&
+      p.mishna === editingParallel?.mishna &&
+      p.lineNumber === editingParallel?.lineNumber
+    );
+    
+    if (indexToReplace === -1 || !editingParallel) {
+      showError('מקבילה מקורית לא נמצאה');
+      return;
+    }
+
+    const originalParallel = parallels[indexToReplace];
+
+    try {
+      // Check if it's a same-mishna parallel
+      const isSameMishna = originalParallel.tractate === tractate && 
+                          originalParallel.chapter === chapter && 
+                          originalParallel.mishna === currentMishna.mishna;
+      
+      if (isSameMishna) {
+        // For same-mishna parallels, handle both sides sequentially
+        await LineService.updateParallel(tractate, chapter, currentMishna.mishna, currentLineNumber, editedParallel);
+        
+        // Small delay to avoid conflicts
+        await new Promise(resolve => setTimeout(resolve, 200));
```

Comment:
why do you need this delay? it's not good to add those unexpected waits. if the previous command has "await" and completes the action - it should be enough